File: src/lessons/welcome/MethodsReturningEntity.java
package lessons.welcome;


public class MethodsReturningEntity extends jlm.universe.bugglequest.SimpleBuggle {
	@Override
	public void forward(int i)  { 
		throw new RuntimeException("forward(int) forbidden in this exercise");
	}

	@Override
	public void backward(int i) {
		throw new RuntimeException("backward(int) forbidden in this exercise");
	}


	@Override
	public void run() { 
		for (int i=0; i<7; i++) {
			if (haveBaggle()) 
				return;
			turnRight();
			forward();
			turnLeft();
		}
	}
	/* BEGIN TEMPLATE */
	public boolean haveBaggle() {
		/* BEGIN SOLUTION */
		boolean res = false;
		for (int i=0; i<6; i++) {
			if (isOverBaggle()) 
				res = true;
			forward();
		}
		for (int i=0; i<6; i++) 
			backward();
		return res;
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/MethodsReturningEntity.py
def forward(i):
	errorMsg("forward(int) forbidden in this exercise")

def backward(i):
	errorMsg("backward(int) forbidden in this exercise")

def run():
	for i in range(7):
		if haveBaggle():
			return
		turnRight()
		forward()
		turnLeft()

# BEGIN TEMPLATE 
def haveBaggle():
    # BEGIN SOLUTION 
	res = False
	for i in range(6):
		if isOverBaggle():
			res = True
		forward()
	for i in range(6):
		backward()
	return res
	# END SOLUTION
#END TEMPLATE

